mfr/providers/osf/provider.py

```diff
@@ -84,6 +84,9 @@ def __init__(self, request, url):
         return streams.ResponseStreamReader(response, unsizable=True)
 
     async def _fetch_download_url(self):
+        # Waterbutler url provided
+        if 'v1/resources' in self.url:
```

Comment:
I think this will pass if we're trying to download a file with `v1/resources` in the filename by the v0 api (possible with non-osfstorage).  e.g. `http://localhost:7777/file?path=/foo/v1/resource-center.txt&provider=googledrive&nid=vjsdy&accept_url=false&mode=render`.  Can you anchor the match to start at the beginning of the url?